test/jdk/java/foreign/TestVarArgs.java

```diff
@@ -192,7 +192,7 @@ private static void check(int index, MemorySegment ptr, List<Arg> args) {
              * address in java on the Big-Endian(BE) platforms such as AIX.
              */
             if (isAixOS && (layout instanceof ValueLayout) && (((ValueLayout)layout).carrier() == float.class)) {
-                MemorySegment doubleSegmt = MemorySegment.ofAddress(ptr, JAVA_DOUBLE.byteSize(), session);
+                MemorySegment doubleSegmt = MemorySegment.ofAddress(ptr.address()).reinterpret(JAVA_DOUBLE.byteSize());
```

Comment:
Is this still needed after [8308031: Linkers should reject unpromoted variadic parameters](https://github.com/ibmruntimes/openj9-openjdk-jdk21/commit/4bfa041ab44656d3f28c78b0bb3ca2cd2c1523f6)?

Reply:
It is hard to determine for the moment given the test suite is not intended for AIX. So I believe it still needed to deal with the conversion on AIX.